src/callback.rs

```diff
@@ -231,33 +232,33 @@ macro_rules! callback_body {
 /// Then this will fail to compile, because the result of #foo borrows _slf, but _slf drops when
 /// the block passed to the macro ends.
 #[doc(hidden)]
-#[macro_export]
-macro_rules! callback_body_without_convert {
-    ($py:ident, $body:expr) => {{
-        let pool = $crate::GILPool::new();
-        let unwind_safe_py = std::panic::AssertUnwindSafe(pool.python());
-        let result = match std::panic::catch_unwind(move || -> $crate::PyResult<_> {
-            let $py = *unwind_safe_py;
-            $body
-        }) {
+pub unsafe fn handle_panic<
+    R: PyCallbackOutput,
+    F: FnOnce(Python) -> crate::PyResult<R> + UnwindSafe,
+>(
```

Comment:
Maybe `where` clause would be better for such long type parameter definitions.

Reply:
good idea